### ISSUE TITLE
Different behavior when using asyncOptions

### ISSUE THREAD

**Issue by tobilg** [joeferner/node-java#305]
I have a project where I initially didn't define the `asyncOptions`. This was for the POC, and led to a somehow quirky code. After proving the chosen implementations worked, I tried to streamline the code by using 

```javascript
java.asyncOptions = {
    asyncSuffix: "Async",
    syncSuffix: "", 
    ifReadOnlySuffix: "_alt"
};
```

to get rid of all the additional `Sync` suffixes in the method names. Turns out that my interface implementation is somehow no longer working after this:

```javascript
java.newProxy('com.mycompany.class', {
    setName: function(name) {
        ...
    }
}
```

Is there any obvious reason for this behavior? I don't receive an error message, the code is just not executed (I tried to debug this, but with no success). Thanks for any hints!

**Comment by joeferner** [comment]
`java.newProxy` shouldn't be affected by `asyncOptions` could you provide a more complete example of what you are trying to do. I tried this and it worked...

```
public interface TestInterface {
    public void run();
}
```

```
public class RunClass {
    public void runIt(TestInterface ti) {
        ti.run():
    }
}
```

```
var java = require('.');
java.classpath.push("test/");

java.asyncOptions = {
    asyncSuffix: "Async",
    syncSuffix: "",
    ifReadOnlySuffix: "_alt"
};

var myProxy = java.newProxy('TestInterface', {
    run: function () {
        console.log('hello');
    }
});

var runInterface = java.newInstanceSync("RunClass");
runInterface.runIt(myProxy);
```

**Comment by tobilg** [comment]
Thanks for the fast answer! I'm trying to implement the [org.apache.mesos.Scheduler](https://github.com/apache/mesos/blob/master/src/java/src/org/apache/mesos/Scheduler.java) interface to be able to create a Mesos framework with Javascript.

I have a running example at [https://github.com/tobilg/mesos-js-framework/](https://github.com/tobilg/mesos-js-framework/blob/master/mesos-js-framework.js) which uses the standard configuration for method names. As soon as I try to switch to the options as written above, it fails.

Thanks again for your help!

**Comment by unitive-jim** [comment]
I wrote the initial implementation of `asyncOptions` but have since left the project where I was using node-java. My memory is a little rusty and I haven't kept up with any refinements Joe or others have done. But with that said, perhaps I can help.

My first guess is that when you tried to make the knife switch to use the new revised async options that you simply missed some cases where you needed to change the suffix. Are you absolutely sure you changed all of your calls to java methods to use the correct conventions?

But also be aware that there are a few special calls whose conventions are not changed by `asyncOptions`. See [Special note about the exported module functions newInstance, callMethod, and callStaticMethod](https://github.com/joeferner/node-java#special-note-about-the-exported-module-functions-newinstance-callmethod-and-callstaticmethod).

**Comment by tobilg** [comment]
@unitive-jim Thanks for your reply as well! I was aware of the notes about the special functions, and I'm quite sure I respected them.

I will check the suffixes again (did so a few times already). Unfortunately, using the debugger for those proxied classes doesn't prove very helpful to me. I only notice that the Scheduler interface code is not executed if I set some breakpoints inside the "no sync" version.
